nsqd/nsqd.go

```diff
@@ -365,6 +399,9 @@ func (n *NSQD) Exit() {
 		n.httpsListener.Close()
 	}
 
+	n.serf.Leave()
```

Comment:
I don't think we want to `Leave` before shutting down - specifically because we want to be able to rejoin as the exact same node if the process was just restarting.

I don't recall the exact nature of how this works in `serf`, so I should probably re-educate myself.